py/picca/fitter2/effective-bins.py

```diff
@@ -0,0 +1,189 @@
+        xi_best_fit = pars['bao_amp']*d.xi_model(k, pk_lin-pksb_lin, pars)
+        pars['SB'] = True
+        snl = pars['sigmaNL_per']
+        pars['sigmaNL_per'] = 0
+        xi_best_fit += d.xi_model(k, pksb_lin, pars)
+        pars['sigmaNL_per'] = snl
+        pars['SB'] = False
+
+    return xi_best_fit
+
+def plot_xi(xi,title=' '):
+    plt.figure()
+    plt.imshow(xi.reshape((np,nt)),origin=0,interpolation='nearest',
+        extent=[rt_min,rt_max,rp_min,rp_max],cmap='seismic',
+        vmin=-max(abs(xi.min()),abs(xi.max())),vmax=max(abs(xi.min()),abs(xi.max())))
+    plt.colorbar()
+    plt.ylabel(r"$r_{\parallel}$",size=20)
+    plt.xlabel(r"$r_{\perp}$",size=20)
+    plt.title(title,size = 20)
+
+def xi_mod_p(x,pname,pars):
+    pars2 = copy.deepcopy(pars)
+    pars2[pname] = x
+    return xi_mod(pars2,dic_init)
+
+def compute_dm_dp(freep,pars):
+    dm = {}
+    for p in freep :
+        g = functools.partial(xi_mod_p,pname=p,pars=pars)
+        dm[p] = derivative(g,pars[p])
+    return dm
+
+def compute_M(dm_dp,icov):
+    M = sp.zeros(icov.shape)
+    for i in range(icov.shape[0]):
+        for j in range(icov.shape[0]):
+            M[i,j] = dm_dp[i] * icov[i,j] * dm_dp[j]
+    return M
+
+def compute_z0(M,z):
+    res = 0
+    den = 0
+    for i in range(z.shape[0]):
+        for j in range(z.shape[0]):
+            res += M[i,j]*z[i]
+            den += M[i,j]
+    return res,den
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+
+    parser.add_argument('--chi2-file', type=str, required=True,
+        help = "Path to the config 'chi2.ini' file used in fitter2")
+
+    parser.add_argument('--cor-name', type=str, required=True,
+        help = "Name of the correlation in 'config.ini' to look at")
+
+    parser.add_argument('--params', type=str,default=[], required=False, nargs='*',
+        help='List of the fitted parameters')
+
+    parser.add_argument('--plot-effective-bins', action='store_true',
+        help='Display an image with the bins involved in the fit of each selected parameter')
+
+    args = parser.parse_args()
+    chi2_file = args.chi2_file
+
+    ### Open files
+    dic_init = fit_parser.parse_chi2(chi2_file)
+    h5_file = dic_init['outfile']
+    free_pars,fixed_pars,best_fit_pars,err_best_fit_pars,xi_best_fit = extract_h5file(h5_file,args.cor_name)
+    for p in args.params:
+        if p not in free_pars:
+            print('ERROR : parameter %s is not fitted'%(p))
+            sys.exit(12)
+    ico,rp,rt,np,nt,rp_min,rp_max,rt_min,rt_max,r_min,r_max,z = extract_data(chi2_file,dic_init)
+
+    ### Computation of the effective bins
+    dm_dp = compute_dm_dp(free_pars,best_fit_pars)
+    apply_mask(dm_dp,ico,z,dic_init)
+
+    for p in args.params:
+        print('Parameter {}'.format(p))
+        M = compute_M(dm_dp[p],ico)
+        res,den = compute_z0(M,z)
+        print('res = {}, den = {}'.format(res,den))
```

Comment:
@vserret, could you be more verbose on what `res` and `den` mean? Maybe use the full name?